test/ibmq/test_ibmq_job.py

```diff
@@ -289,12 +292,87 @@ def test_retrieve_job_error(self, provider):
     def test_retrieve_jobs_status(self, provider):
         """Test retrieving jobs filtered by status."""
         backend = provider.get_backend('ibmq_qasm_simulator')
-        job_list = provider.backends.jobs(backend_name=backend.name(),
-                                          limit=5, skip=0, status=JobStatus.DONE)
+        # Get the most recent jobs that are done.
+        status_args = [JobStatus.DONE, 'DONE', [JobStatus.DONE], ['DONE']]
+        for arg in status_args:
+            with self.subTest(arg=arg):
+                backend_jobs = backend.jobs(limit=10, skip=0, status=arg)
+                self.assertTrue(backend_jobs)
+
+                for job in backend_jobs:
+                    self.assertTrue(job.status() is JobStatus.DONE,
+                                    "job status for job {} was '{}' but "
+                                    "it should be: '{}'"
+                                    .format(job.job_id(), job.status(), JobStatus.DONE))
 
-        self.assertTrue(job_list)
-        for job in job_list:
-            self.assertTrue(job.status() is JobStatus.DONE)
+    @requires_provider
+    def test_retrieve_all_job_statuses(self, provider):
+        """Test retrieving jobs filtered by all possible job statuses."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+        # pylint: disable=unnecessary-comprehension
+        all_job_statuses = [status for status in JobStatus]
+        # Get the most recent jobs with any job status.
+        backend_jobs_filtered = backend.jobs(limit=10, status=all_job_statuses)
+        backend_jobs = backend.jobs(limit=10)
+        self.assertTrue(backend_jobs)
+        self.assertTrue(backend_jobs_filtered)
+
+    @requires_provider
+    def test_retrieve_multiple_job_statuses(self, provider):
```

Comment:
I think the multi-status filtering (this and the next tests) would be much more valuable if we can _confirm_ they return jobs in all of the specified statuses. For example, if only `CANCELLED` jobs are returned, it's hard to tell if no `ERROR` jobs were found or if the query was constructed incorrectly. To accomplish this, you would need to actually create a cancelled and a failed jobs. It might be easier to combine these two tests (using `subTest`) so you only need to create those jobs once.

Reply:
The existing code says the argument filters win:
```
        if db_filter:
            # Argument filters takes precedence over db_filter for same keys
            api_filter = {**db_filter, **api_filter}
```